src/buza/views.py

```diff
@@ -69,3 +75,131 @@ def edit(request: HttpRequest) -> HttpResponse:
 
 class UserDetail(generic.DetailView):
     model = models.User
+
+
+class QuestionDetail(generic.DetailView):
+    model = models.Question
+
+
+class QuestionList(generic.ListView):
+    model = models.Question
+    ordering = ['-created']
+
+
+class QuestionCreate(LoginRequiredMixin, generic.CreateView):
+    model = models.Question
+    fields = [
+        'title',
+        'body',
+    ]
+
+    def form_valid(self, form: ModelForm) -> HttpResponse:
+        """
+        Set the question's author to the posting user.
+        """
+        question: models.Question = form.instance
+        author: models.User = self.request.user
+        assert author.is_authenticated, author
+        question.author = author
+        return super().form_valid(form)
+
+    def get_success_url(self) -> str:
+        """
+        Redirect to the question.
+        """
+        question: models.Question = self.object
+        success_url: str = reverse('question-detail', kwargs=dict(pk=question.pk))
+        return success_url
+
+
+class QuestionUpdate(LoginRequiredMixin, generic.UpdateView):
+    model = models.Question
+    fields = [
+        'title',
+        'body',
+    ]
+    question: models.Question
+    template_name_suffix = '_update_form'
```

Comment:
We shouldn't need to override the template name or suffix here: the default `question_form.html` should be fine, and we can share it between the create and update forms.

Reply:
Thanks.
I just wanted to add the update question button. Adding a whole template seems dramatic

Reply:
To conditionalise parts of the form to appear differently for create versus update, the idiomatic way to do it is to test for `question.pk`: it will be set for existing objects, and `None` for new objects being created.

For example, for button text:

```
{% if question.pk %}
    Update question
{% else %}
    Ask question
{% endif%}
```